### PR TITLE
[DR-OCR] Integration test for DirectRequest OCR

### PR DESCRIPTION
**Pull request by bolekk**
4 nodes on a simulated chain handling 20 requests with batch limit of 5.

### CODE REVIEW

**Review by KuphJr — DISMISSED** [review]
I went through this and everything looks good, but you should probably have @pinebit approve before merging given my limited Go experience

File: core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go [hunk 1]
```diff
@@ -0,0 +1,307 @@
...
+
+type deployedClientContract struct {
+	Address  common.Address
+	Contract *ocr2dr_client_example.OCR2DRClientExample
+}
+
+func StartNewChainWithContracts(t *testing.T, nClients int) (*bind.TransactOpts, *backends.SimulatedBackend, *time.Ticker, common.Address, *ocr2dr_oracle.OCR2DROracle, []deployedClientContract) {
+	owner := testutils.MustNewSimTransactor(t)
+	sb := new(big.Int)
+	sb, _ = sb.SetString("100000000000000000000", 10) // 1 eth
```
Comment (pinebit) [inline core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go:118]: `assets.NewEth(1)`

**Review by pinebit — DISMISSED** [review]

File: core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go (hunk 1 above)
Comment (bolekk) [reply core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go:118]: It doesn't work. I need wei but NewEth doesn't convert to wei (it converts to 1).

File: core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+func StartNewMockEA(t *testing.T) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		b, err := io.ReadAll(req.Body)
+		require.NoError(t, err)
+		var jsonMap map[string]any
+		require.NoError(t, json.Unmarshal(b, &jsonMap))
+		source := jsonMap["data"].(map[string]any)["source"].(string)
+		res.WriteHeader(http.StatusOK)
+		// prepend "0xab" to source and return as result
+		res.Write([]byte(fmt.Sprintf(`{"data": {"result": "0xab%s", "error": ""}}`, source)))
```
[the remark this answers is not in the document]
Comment (bolekk) [reply core/services/ocr2/plugins/directrequestocr/test/internal/testutils.go:305]: that's ok

**Pull request merged**